test/processorproxy_test.cc

```diff
@@ -67,11 +69,8 @@ TEST(ProcessorProxySuccessTest, ProcessWorks) {
     EXPECT_EQ(grpc::StatusCode::OK, status.error_code());
     EXPECT_EQ(1, report.size());
     EXPECT_EQ("hop", report.begin()->get_string_data());
-    std::string ns_str;
-    for (auto const& seg : report.begin()->ns()) {
-        ns_str += "/" + seg.value;
-    }
-    EXPECT_EQ("/foo/bar", ns_str);
+    std::string ns_str = extract_ns(*report.begin());
+    EXPECT_EQ("/foo/bar", extract_ns(*report.begin()));
```

Comment:
It seems that here you can write 
` EXPECT_EQ("/foo/bar", ns_str);` instead of calling again `extract_ns()` or you can remove the `ns_str` variable.